File: java/sodium/src/sodium/BehaviorSink.java
package sodium;

public final class BehaviorSink<A> extends Behavior<A> {
    public BehaviorSink(A initValue) {
    	super(new EventSink<A>(), initValue);
    }
    
    public void send(A a)
    {
        ((EventSink<A>)event).send(a);
    }
}

File: c#/Sodium/src/Sodium/BehaviorSink.cs
namespace Sodium
{

  public class BehaviorSink<TA> : Behavior<TA>
  {
    public BehaviorSink(TA initValue)
      : base(new EventSink<TA>(), initValue)
    {
    }

    public void Send(TA a)
    {
      ((EventSink<TA>)Event).Send(a);
    }
  }

}
